### DISCUSSION TITLE
Using GitHub api to upload multiple files

### DISCUSSION THREAD

**Discussion by Karan-S-dev** [community/community#29632] [Apps, API and Webhooks]
Hi,

I have done creating file in my repo using github api. Using the following example,

curl \
  -X PUT \
  -H "Accept: application/vnd.github+json" \ 
  -H "Authorization: token <TOKEN>" \
  https://api.github.com/repos/OWNER/REPO/contents/PATH \
  -d '{"message":"my commit message","committer":{"name":"Monalisa Octocat","email":"octocat@github.com"},"content":"bXkgbmV3IGZpbGUgY29udGVudHM="}'

This is working. But is there anyway to upload multiple files at the same time using GitHub api?

**Comment by lbirkert** [comment]
That is not possible, at least not in one commit I'm afraid. Why don't you clone the repository change the files you want to change and then commit and push? It would help a lot if you tried to explain what you are trying to do, and why you want to upload files using the api.

Best regards

**Reply by Karan-S-dev** [reply]
I have built a documentation site using docusaurus. Now we are trying to automate everything. What basically docusaurus does is if you upload any md file in the github repository, then it will automatically create a web page with the contents in that  md file. 

I have implemented this for single file. I have a upload button in my site and when a customer uploads a md file in that button, then it will automatically add that file in the github repo using github api. 

But What if the customer wants to upload more that 1 md file, then it becomes trickier for me?  Thats why i wanted to know if there is a option to upload multiple files using github api. It its not possible  means, then i can look for other solutions.

Thanks for your time.
